index.d.ts

```diff
@@ -0,0 +1,28 @@
+export interface AppAuthInput {
+    scopes: string[];
+    issuer: string;
+    clientId: string;
+    redirectUrl: string;
+    aditionalParameters?: object;
+  }
+
+  export interface AuthorizeResult {
+    accessToken: string;
+    accessTokenExpirationDate: string;
+    additionalParameters: object;
+    idToken: string;
+    refreshToken: string;
+    tokenType: string;
+  }
+
+  export function authorize(properties: AppAuthInput): Promise<AuthorizeResult>;
+
+  export function refresh(
+    properties: AppAuthInput,
+    { refreshToken: string }
+  ): Promise<AuthorizeResult>;
+
+  export function revoke(
+    properties: AppAuthInput,
+    { tokenToRevoke: string, sendClientId: boolean }
```

Comment:
- The only properties needed for `revoke` are `clientId` and `issuer`. It doesn't matter if you pass in the whole configuration object, but it's not all used. I'm not sure if there's an easy way to reflect that in TypeScript?
- `sendClientId` is optional, it defaults to `false`

Reply:
Thanks for the feedback!
As a proposal I made the `clientId` and `issues` in a separate interface that the `AuthConfiguration` extends, making it 1 definition. 
If we don't extend it'll mean we'd need to have two instances of the config object.

`sendClientId` made optional